### PR TITLE
Fix failing test:unit pipeline

### PR DESCRIPTION
**Pull request by VSaric**
This PR fixes failing `test:unit` on pipeline. Added some more checks to the `edit-gas-display.component.js` to ensure that errors are not thrown in case `gasLimit` or `properGasLimit` are not defined.

### CODE REVIEW

**Review by danjm — DISMISSED** [review]
LGTM, we can merge this once the other failure is passing

**Review by digiwand — DISMISSED** [review]
LGTM! I wonder if we would like to specify that this is an optional prop in `propTypes`

File: ui/components/app/edit-gas-display/edit-gas-display.component.js
Comment (mcmire) [inline ui/components/app/edit-gas-display/edit-gas-display.component.js:114]: Is `gasLimit` a number or a hex value here? Just wondering whether it would make more sense to say `if (Boolean(gasLimit) && Boolean(properGasLimit) && ...)` here.

Comment (VSaric) [reply ui/components/app/edit-gas-display/edit-gas-display.component.js:114]: @mcmire `gasLimit` is a number. I tried with this changes that you suggested here `(Boolean(gasLimit) && Boolean(properGasLimit) && ...)`, unit test and coverage are passed also.
@danjm Can you please look at this and comment what do you think about it? I can add this changes if you agree with this.

**Comment by brad-decker** [comment]
@VSaric I Have rebased this onto a branch that hopefully doesn't have the test audit failure. This PR cannot merge until the update-nanoid pr is merged then we'll rebase this on the other branch and finally merge into develop. Assuming @mcmire is satisfied with your response on the feedback

File: ui/components/app/edit-gas-display/edit-gas-display.component.js
Comment (danjm) [reply ui/components/app/edit-gas-display/edit-gas-display.component.js:114]: @mcmire for which case does the `Boolean` call make a difference?

Comment (mcmire) [reply ui/components/app/edit-gas-display/edit-gas-display.component.js:114]: `Boolean(...)` would check if the value in question is null or undefined. Without that, if `gasLimit` is 0, but is less than `properGasLimit`, then we wouldn't show a warning. Seems like we might not want to do that.

Comment (danjm) [reply ui/components/app/edit-gas-display/edit-gas-display.component.js:114]: `Boolean(0) === false` right?

So do you mean that we need to do something like `gasLimit || gasLimit === 0`... or, what if we do `parseInt(gasLimit) !== NaN`?

Comment (danjm) [reply ui/components/app/edit-gas-display/edit-gas-display.component.js]: I just pushed a commit that I think resolves the concern

Comment (mcmire) [reply ui/components/app/edit-gas-display/edit-gas-display.component.js]: Yup, that looks good!

**Review by brad-decker — APPROVED** [review]

**Review by mcmire — APPROVED** [review]

**Pull request merged**
